sif_parser/sif_open.py

```diff
@@ -150,3 +151,77 @@ def xr_open(sif_file, ignore_corrupt=False, lazy=None):
+    """
+    Read from directory binary files and metadata and return a np.array.
+    
+    Parameters
+    ----------
+    spool_dir: 
+        path to the directory containing the spooling files. 
+        Must contain at least the following set of files:
+        - sifx_file: 1 file with the extention "*.sifx". This is the header of the file containing metadata.
+        - ini_file: 1 file with the extension "
+        - spooled_files: file or set of files with the extention "*spool.dat" containing the imgae data as binary files.
+    ignore_missing: 
+        True if ignore mising binary files.
+    """
+    dat_files_list = sorted(glob.glob(spool_dir + "/*spool.dat" ))
+    ini_file = glob.glob(spool_dir + "/*.ini" )
+    sifx_file = glob.glob(spool_dir + "/*.sifx")
+
+    if len(dat_files_list) < 1:
+        raise ValueError('Not Binary file(s) with extension {} found in the directory provided {} '.format(
+            "*spool.dat", spool_dir))
+    if len(ini_file) < 1:
+        raise ValueError('Not "ini_file" file with extension {} found in the directory provided {} '.format(
+            "*.ini" , spool_dir))
+    if len(sifx_file) < 1:
+        raise ValueError('Not "sifx_file" file with extension {} found in the directory provided {} '.format(
+            "*.sifx", spool_dir))
+
+
+
+    with open(ini_file[0], "r") as f:
+
+        lines = f.readlines()
+
+    ini_info = OrderedDict([line[:-1].replace(" ", "").split("=") for line in lines[1:4]])
+    
+    # read only metadata (ignoring expected warning on missing data)
+    with warnings.catch_warnings():
+        warnings.simplefilter("ignore")
+        _ , metadata = np_open(sifx_file[0], ignore_corrupt=True)
+
+    # get the actual shape of the image from metadata
+    x, y = metadata["DetectorDimensions"]
+    t = metadata["NumberOfFrames"]
+
+    if len(dat_files_list) != t:
+        if not ignore_missing:
+            raise ValueError('The spooling acquisition might be corrupt. Number of files should be {} '
+                        'according to the header, but only {} binary files were found in the directory.'.format(
+                            t, len(dat_files_list)))
+        else:
+            warnings.warn('The spooling acquisition might be corrupt. Number of files should be {} '
+                        'according to the header, but only {} binary files were found in the directory.'.format(
+                            t, len(dat_files_list)))
+            t = len(dat_files_list)
+
+    # shape of ini file
+    x_, y_ =  int( int(ini_info['AOIStride']) / 2 ), int(ini_info['AOIHeight'])
+   # account for the extra padding to trim it later
+    end_padding =  x_ - x
+    
+    # create np array with the given info     
+    data = np.empty( [t, y_, x_] ) 
+
+    for frame in range(t):
+        data[frame, ...] = np.fromfile(dat_files_list[frame], 
+                                       offset=0, 
+                                       dtype=np.uint16, 
```

Comment:
I think this dtype may depend on PixelEncoding in the .ini file. We may raise an error when we encounter an encoding other than Mono16.